app/src/main/java/com/bogdan/codeforceswatcher/redux/middlewares/NotificationMiddleware.kt

```diff
@@ -0,0 +1,96 @@
+package com.bogdan.codeforceswatcher.redux.middlewares
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.Color
+import androidx.annotation.RequiresApi
+import androidx.core.app.NotificationCompat
+import com.bogdan.codeforceswatcher.CwApp
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.activity.MainActivity
+import com.bogdan.codeforceswatcher.feature.users.redux.request.UsersRequests
+import org.rekotlin.Middleware
+import org.rekotlin.StateType
+
+val notificationMiddleware: Middleware<StateType> = { _, _ ->
+    { next ->
+        { action ->
+
+            if (action is UsersRequests.FetchUsers.Success) {
+                if (!action.isUser) {
+                    var notificationText = ""
+                    var flag = 0
+
+                    for (ratingChange in action.result) {
+                        if (flag == 1) {
+                            notificationText += "\n"
+                        }
+                        flag = 1
+                        notificationText += ratingChange.first + " " + if (ratingChange.second < 0) {
+                            "${ratingChange.second}"
+                        } else {
+                            "+${ratingChange.second}"
+                        }
+                    }
+
+                    if (notificationText.isNotEmpty()) {
```

Comment:
Can be separated to function, which forms notification and one which checks that result is not empty and shows it.